tests/unit/session/test_parallel.py

```diff
@@ -189,3 +189,113 @@ def test_parallel_show_output(cmd, initproj, monkeypatch):
     assert "stderr env" not in result.out, result.output()
     assert "stdout always" in result.out, result.output()
     assert "stderr always" in result.out, result.output()
+
+
+def test_parallel_result_json(cmd, initproj, tmpdir):
+    import json
+
+    initproj(
+        "pkg123-0.7",
+        filedefs={
+            "tox.ini": """
+            [tox]
+            envlist = a, b
+            isolated_build = true
+            [testenv]
+            commands=python -c "import sys; print(sys.executable)"
+        """,
+            "pyproject.toml": """
+            [build-system]
+            requires = ["setuptools >= 35.0.2"]
+            build-backend = 'setuptools.build_meta'
+                        """,
+        },
+    )
+    serial_result_json = tmpdir.join("serial.json")
+    result = cmd("--result-json", "{}".format(serial_result_json))
+    assert serial_result_json.check(file=True)
+    with serial_result_json.open("rb") as f:
+        serial_data = json.loads(f.read().decode("utf-8"))
+    assert "setup" in serial_data["testenvs"]["a"]
+    assert "test" in serial_data["testenvs"]["a"]
+    assert "setup" in serial_data["testenvs"]["b"]
+    assert "test" in serial_data["testenvs"]["b"]
+    parallel_result_json = tmpdir.join("parallel.json")
+    result = cmd("--parallel", "all", "--result-json", "{}".format(parallel_result_json))
+    result.assert_success()
+    assert parallel_result_json.check(file=True)
+    with parallel_result_json.open("rb") as f:
+        parallel_data = json.loads(f.read().decode("utf-8"))
+    assert "setup" in parallel_data["testenvs"]["a"]
+    assert "test" in parallel_data["testenvs"]["a"]
```

Comment:
wouldn't be a better test to assert the serial is the same as the parallel (maybe just cleanup timestamps and hashseed)🤔

Reply:
That felt too fragile, as the output depends on external tools.